sdk/go/pulumi-language-go/main.go

```diff
@@ -157,10 +160,71 @@ func newLanguageHost(engineAddress, tracing string) pulumirpc.LanguageRuntimeSer
 	}
 }
 
+// modInfo is the useful portion of the output from `go list -m -json all`
+// with respect to plugin acquisition
+type modInfo struct {
+	Path    string
+	Version string
+}
+
+func getPluginName(modName string) string {
+	// github.com/pulumi/pulumi-aws/sdk/... => aws
+	pluginPart := strings.Split(modName, "/")[2]
+	return strings.SplitN(pluginPart, "-", 2)[1]
+}
+
 // GetRequiredPlugins computes the complete set of anticipated plugins required by a program.
+// We're lenient here as this relies on the `go list` command and the use of modules.
+// If the consumer insists on using some other form of dependency management tool like
+// dep or glide, the list command fails with "go list -m: not using modules"
 func (host *goLanguageHost) GetRequiredPlugins(ctx context.Context,
```

Comment:
Can we add any test coverage of this?

Reply:
I refactored to test the code to test the filtering and conversion of the module result from `go list` into a plugin description. 

This seems sufficient to me. Any further validation would require an integration test with a dependency on pulumi-aws.